// stepper-library/esm5/lib/stepset/step/step.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/stepset/step/step.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, ContentChildren } from '@angular/core';
import { StepsetComponent } from '../stepset.component';
import { NgForm } from '@angular/forms';
var StepComponent = /** @class */ (function () {
    function StepComponent(stepset) {
        this.stepset = stepset;
        this.active = false;
        this.valid = false;
    }
    /**
     * @return {?}
     */
    StepComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @return {?}
     */
    StepComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        this.form = this.form.toArray();
        this.stepset.addStep(this);
    };
    StepComponent.decorators = [
        { type: Component, args: [{
                    selector: 'step',
                    template: "<div *ngIf=\"active\" [ngClass]=\"active ? 'slide' : ''\">\n    <h5 class=\"title\">{{title}}</h5>\n    <ng-content></ng-content>\n</div>",
                    styles: [":host{width:100%}.slide{-webkit-transform:translateX(100%);transform:translateX(100%);-webkit-animation:.5s forwards animateStep;animation:.5s forwards animateStep}@-webkit-keyframes animateStep{0%{-webkit-transform:translateX(100%);transform:translateX(100%)}100%{-webkit-transform:translateX(0);transform:translateX(0)}}@keyframes animateStep{0%{-webkit-transform:translateX(100%);transform:translateX(100%)}100%{-webkit-transform:translateX(0);transform:translateX(0)}}.title{font-size:1.5em}@media (max-width:426px){.title{display:none}}"]
                }] }
    ];
    /** @nocollapse */
    StepComponent.ctorParameters = function () { return [
        { type: StepsetComponent }
    ]; };
    StepComponent.propDecorators = {
        title: [{ type: Input }],
        form: [{ type: ContentChildren, args: [NgForm,] }]
    };
    return StepComponent;
}());
export { StepComponent };
if (false) {
    /** @type {?} */
    StepComponent.prototype.active;
    /** @type {?} */
    StepComponent.prototype.valid;
    /** @type {?} */
    StepComponent.prototype.title;
    /** @type {?} */
    StepComponent.prototype.form;
    /**
     * @type {?}
     * @private
     */
    StepComponent.prototype.stepset;
}
