// stepper-library/esm5/lib/stepset/stepset.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/stepset/stepset.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Output, Input, ElementRef, Renderer2, ViewChild } from '@angular/core';
var StepsetComponent = /** @class */ (function () {
    function StepsetComponent(rd) {
        this.rd = rd;
        this.steps = [];
        this.stepNo = 0;
        this.totalSteps = 0;
        this.btnPos = 'btn-left';
        this.next = new EventEmitter();
        this.submitEvent = new EventEmitter();
        this.radius = 35;
        this.strokeDashOffset = Math.PI * 2 * this.radius;
    }
    /**
     * @return {?}
     */
    StepsetComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @return {?}
     */
    StepsetComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        this.steps[0].active = true;
    };
    /**
     * @param {?} step
     * @return {?}
     */
    StepsetComponent.prototype.addStep = /**
     * @param {?} step
     * @return {?}
     */
    function (step) {
        this.totalSteps = this.steps.push(step);
    };
    /**
     * @param {?} i
     * @return {?}
     */
    StepsetComponent.prototype.jumpToStep = /**
     * @param {?} i
     * @return {?}
     */
    function (i) {
        if (this.steps[i > 0 ? i - 1 : i].valid) {
            this.stepNo = i;
            this.setActiveStep(this.stepNo);
        }
    };
    /**
     * @return {?}
     */
    StepsetComponent.prototype.nextStep = /**
     * @return {?}
     */
    function () {
        if (this.steps[this.stepNo].form[0]['valid']) {
            this.steps[this.stepNo].valid = true;
            this.stepNo < this.totalSteps ? this.setActiveStep(++this.stepNo) : null;
            this.next.emit(this.steps[this.stepNo].valid);
        }
    };
    /**
     * @return {?}
     */
    StepsetComponent.prototype.setProgressbarColor = /**
     * @return {?}
     */
    function () {
        this.progressValue = Math.floor(100 / (this.totalSteps - 1)) * this.stepNo - 100;
        this.rd.setStyle(this.circularBar.nativeElement, 'stroke-dashoffset', ((100 - this.progressValue) / 100) * (Math.PI * (2 * this.radius)) - this.strokeDashOffset);
    };
    /**
     * @return {?}
     */
    StepsetComponent.prototype.prevStep = /**
     * @return {?}
     */
    function () {
        this.stepNo > 0 ? this.setActiveStep(--this.stepNo) : null;
    };
    /**
     * @param {?} index
     * @return {?}
     */
    StepsetComponent.prototype.setActiveStep = /**
     * @param {?} index
     * @return {?}
     */
    function (index) {
        this.setProgressbarColor();
        this.steps.forEach((/**
         * @param {?} s
         * @return {?}
         */
        function (s) {
            s.active = false;
        }));
        this.steps[index].active = true;
    };
    /**
     * @return {?}
     */
    StepsetComponent.prototype.submit = /**
     * @return {?}
     */
    function () {
        this.submitEvent.emit(true);
    };
    StepsetComponent.decorators = [
        { type: Component, args: [{
                    selector: 'stepset',
                    template: "<div id=\"stepset-container\">\n  <div id=\"lg-progress\">\n    <div id=\"step-nav\">\n      <div *ngFor=\"let s of steps;let i=index\" (click)=\"jumpToStep(i)\" [ngClass]=\"i < stepNo ? 'validStep' : ''\">\n        {{i+1}}<p>{{s.title}}</p>\n      </div>\n      <svg id=\"line-progress\" height=7>\n        <g id=\"progress-container\">\n          <line x1=\"0\" y1=\"50%\" x2=\"100%\" y2=\"50%\" stroke-width=7 />\n        </g>\n        <g id=\"progress-content\">\n          <line x1=\"0\" y1=\"50%\" x2=\"100%\" y2=\"50%\" stroke-width=7 fill=\"transparent\" stroke-dasharray=\"0%\"\n            stroke-dashoffset=\"0%\" #progressbar [ngStyle]=\"{'transform':'translateX('+ progressValue +'%)'}\" />\n        </g>\n      </svg>\n    </div>\n  </div>\n\n  <div id=\"xs-progress\">\n    <span class=\"step-display\">{{this.stepNo+1}} of {{this.totalSteps}}</span>\n    <svg id=\"svg\" width=\"100\" height=\"100\">\n      <circle [attr.r]='radius' cx=\"50\" cy=\"50\" fill=\"transparent\" [attr.stroke-dasharray]=\"strokeDashOffset\"\n        stroke-dashoffset=\"0\"></circle>\n      <circle #bar id=\"bar\" [attr.r]='radius' cx=\"50\" cy=\"50\" fill=\"transparent\"\n        [attr.stroke-dasharray]=\"strokeDashOffset\" [attr.stroke-dashoffset]='strokeDashOffset'></circle>\n    </svg>\n    <h3 class=\"title\">\n      {{steps[stepNo]?.title}}\n    </h3>\n  </div>\n  <div id=\"content-container\">\n    <ng-content></ng-content>\n    <hr>\n    <div id=\"buttons\" [ngClass]=\"btnPos\">\n      <button class=\"prev-btn\" (click)=\"prevStep()\">Back</button>\n      <button class=\"next-btn\" (click)=\"nextStep()\" *ngIf=\"totalSteps > stepNo+1\">Next</button>\n      <button (click)=\"submit()\" [ngStyle]=\"styles\" *ngIf=\"totalSteps == stepNo+1\">Finish</button>\n    </div>\n  </div>\n</div>",
                    styles: ["#stepset-container{padding:10px;box-shadow:0 0 5px rgba(0,0,0,.2);margin:20px;border-radius:4px}#buttons{margin-top:20px;position:relative;display:-webkit-box;display:flex}#buttons button{font-size:1.1em;border:1px solid #ccc;background:#fff;border-radius:4px;padding:10px 15px;margin-right:10px;box-shadow:2px 4px 5px rgba(0,0,0,.2)}#buttons button:focus{outline:0}#buttons button:hover{background-color:#f7f7f7}.btn-left{-webkit-box-pack:left;justify-content:left}.btn-apart{-webkit-box-pack:justify;justify-content:space-between}.btn-right{-webkit-box-pack:end;justify-content:flex-end}.btn-center{-webkit-box-pack:center;justify-content:center}#step-nav{display:-webkit-box;display:flex;-webkit-box-pack:justify;justify-content:space-between;position:relative}#step-nav p{line-height:1.4;padding-top:6px;display:-webkit-box;display:flex;-webkit-box-pack:center;justify-content:center}#step-nav div{cursor:pointer;display:inline-block;width:40px;height:40px;line-height:3;text-align:center;border-radius:50%;z-index:1;background:#ccc;position:relative}#step-nav div:after,#step-nav div:before{position:absolute;left:0;top:0;display:inline-block;border-radius:50%}#step-nav div:after{content:\"\";width:50px;height:50px;background-color:transparent;border:6px solid #fff;-webkit-transform:translate(-10%,-10%);transform:translate(-10%,-10%)}#step-nav div:before{content:\"\\2713\";width:inherit;background-color:#1b9e0b;color:#fff;opacity:0;-webkit-transition:opacity 1s;transition:opacity 1s}#step-nav div.validStep:before{opacity:1}#progress-container{stroke:#ccc}#progress-content{stroke:green}#progress-content line{-webkit-transform:translateX(-100%);transform:translateX(-100%);-webkit-transition:.6s cubic-bezier(.39,.575,.565,1);transition:.6s cubic-bezier(.39,.575,.565,1)}#line-progress{position:absolute;width:100%;top:16px}#svg{stroke-dashoffset:0;stroke:#ccc;stroke-width:5px}#svg #bar{stroke:green;-webkit-transition:stroke-dashoffset 1s;transition:stroke-dashoffset 1s}.title{margin-top:-10px;display:-webkit-box;display:flex;-webkit-box-pack:center;justify-content:center;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-flow:column;margin-left:18px;font-size:1.5em;line-height:1.2}#content-container{padding:10px}.step-display{position:absolute;-webkit-transform:translate(100%,200%);transform:translate(100%,200%);left:-1%}@media (min-width:768px){#content-container{padding:20px 30px;margin-top:40px}#xs-progress{display:none}#lg-progress{display:block;padding:10px 35px}}@media (max-width:426px){#xs-progress{display:inline-block;position:relative;display:-webkit-box;display:flex}#lg-progress{display:none}}"]
                }] }
    ];
    /** @nocollapse */
    StepsetComponent.ctorParameters = function () { return [
        { type: Renderer2 }
    ]; };
    StepsetComponent.propDecorators = {
        progressbar: [{ type: ViewChild, args: ['progressbar',] }],
        styles: [{ type: Input, args: ['finishBtn',] }],
        btnPos: [{ type: Input, args: ['btnPos',] }],
        next: [{ type: Output }],
        circularBar: [{ type: ViewChild, args: ['bar',] }],
        submitEvent: [{ type: Output }]
    };
    return StepsetComponent;
}());
export { StepsetComponent };
if (false) {
    /** @type {?} */
    StepsetComponent.prototype.steps;
    /** @type {?} */
    StepsetComponent.prototype.stepNo;
    /** @type {?} */
    StepsetComponent.prototype.totalSteps;
    /** @type {?} */
    StepsetComponent.prototype.progressValue;
    /** @type {?} */
    StepsetComponent.prototype.strokeDashOffset;
    /** @type {?} */
    StepsetComponent.prototype.progressbar;
    /** @type {?} */
    StepsetComponent.prototype.styles;
    /** @type {?} */
    StepsetComponent.prototype.btnPos;
    /** @type {?} */
    StepsetComponent.prototype.next;
    /** @type {?} */
    StepsetComponent.prototype.circularBar;
    /** @type {?} */
    StepsetComponent.prototype.submitEvent;
    /** @type {?} */
    StepsetComponent.prototype.radius;
    /**
     * @type {?}
     * @private
     */
    StepsetComponent.prototype.rd;
}
